File: core/core-domain/src/main/java/com/nmh/commerce/coupon/DurationPeriodPolicy.java
package com.nmh.commerce.coupon;

import lombok.Getter;

import java.time.Duration;
import java.time.LocalDateTime;

@Getter
public class DurationPeriodPolicy implements ExpirationPeriodPolicy {
    private final Duration duration;

    public DurationPeriodPolicy(Duration duration) {
        this.duration = duration;
    }

    @Override
    public void verify(LocalDateTime issuedAt) {
        if (LocalDateTime.now().isAfter(issuedAt.plus(duration))) {
            throw new IllegalArgumentException("쿠폰이 만료되었습니다.");
        }
    }
}

File: core/core-domain/src/main/kotlin/com/nmh/commerce/coupon/DurationPeriodPolicy.kt
package com.nmh.commerce.coupon

import java.time.Duration
import java.time.LocalDateTime

data class DurationPeriodPolicy(
    val duration: Duration
) : ExpirationPeriodPolicy {
    override fun verify(issuedAt: LocalDateTime) {
        require(!LocalDateTime.now().isAfter(issuedAt.plus(duration))) { "쿠폰이 만료되었습니다." }
    }
}
